File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.py
class Solution:
    def findMin(self, nums: List[int]) -> int:
        
        lo = 0
        hi = len(nums)-1
        
        result = nums[lo]
        
        while (lo <= hi):
            if nums[lo] <= nums[hi]:
                result = min(result, nums[lo])
                break
            
            mid = (lo+hi) // 2
            result = min(nums[mid], result)
            
            if nums[lo] <= nums[mid]:
                lo = mid+1
            else:
                hi = mid-1
        
        return result

File: 0153-find-minimum-in-rotated-sorted-array/0153-find-minimum-in-rotated-sorted-array.cpp
class Solution {
public:
    int findMin(vector<int>& nums) {
        
        
        int lo = 0;
        int hi = nums.size()-1;
        
        int result = nums[lo];
        
        while(lo <= hi) {
            
            if (nums[lo] < nums[hi]) {
                return min(result, nums[lo]);
            }
            
            
            int mid = (lo+hi) / 2;
            
            result = min(result, nums[mid]);
            
            if (nums[lo] <= nums[mid]) {
                lo = mid+1;
            }
            else {
                hi = mid-1;
            }
        }
        
        return result;
    }
};